src/sentry/snuba/metrics_layer/query.py

```diff
@@ -0,0 +1,85 @@
+from typing import Any, Dict, Optional, Sequence
+
+from snuba_sdk import Request
+from snuba_sdk.metrics_query import MetricsQuery
+
+from sentry.models import Project
+from sentry.sentry_metrics.utils import string_to_use_case_id
+from sentry.snuba.metrics.fields.base import MetricExpression, RawMetric, metric_object_factory
+from sentry.snuba.metrics.naming_layer.mapping import get_mri, get_public_name_from_mri
+from sentry.snuba.metrics.utils import to_intervals
+
+
+def run_query(request: Request, tenant_ids: Optional[Dict[str, Any]] = None) -> None:
```

Comment:
tenant_ids are part of the request object. Why do you need to include them in this signature?

Reply:
A request was made from the COGS project to set `use_case_id` into the tenant_id. This probably could take place when the request is first built; however, there isn't a repeatable way to do that just yet.

Reply:
I don't think that answers the question. You are modifying `tenant_ids` below, why not modify `request.tenant_ids` instead?

Reply:
you're right, updated.